File: client/clojure/src/cdisc_library_api/specs/export_sdtm_dataset_variables_row.clj
(ns cdisc-library-api.specs.export-sdtm-dataset-variables-row
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def export-sdtm-dataset-variables-row-data
  {
   (ds/opt :Version) string?
   (ds/opt :VariableOrder) string?
   (ds/opt :Class) string?
   (ds/opt :DatasetName) string?
   (ds/opt :VariableName) string?
   (ds/opt :VariableLabel) string?
   (ds/opt :Type) string?
   (ds/opt :ControlledTermsCodelistorFormat) string?
   (ds/opt :Role) string?
   (ds/opt :RoleDescription) string?
   (ds/opt :Description) string?
   })

(def export-sdtm-dataset-variables-row-spec
  (ds/spec
    {:name ::export-sdtm-dataset-variables-row
     :spec export-sdtm-dataset-variables-row-data}))

File: src/main/scala/org/openapitools/client/api/ExportSdtmDatasetVariablesRow.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ExportSdtmDatasetVariablesRow._

case class ExportSdtmDatasetVariablesRow (
  version: Option[String],
variableOrder: Option[String],
`class`: Option[String],
datasetName: Option[String],
variableName: Option[String],
variableLabel: Option[String],
`type`: Option[String],
controlledTermsCodelistOrFormat: Option[String],
role: Option[String],
roleDescription: Option[String],
description: Option[String])

object ExportSdtmDatasetVariablesRow {
  import DateTimeCodecs._

  implicit val ExportSdtmDatasetVariablesRowCodecJson: CodecJson[ExportSdtmDatasetVariablesRow] = CodecJson.derive[ExportSdtmDatasetVariablesRow]
  implicit val ExportSdtmDatasetVariablesRowDecoder: EntityDecoder[ExportSdtmDatasetVariablesRow] = jsonOf[ExportSdtmDatasetVariablesRow]
  implicit val ExportSdtmDatasetVariablesRowEncoder: EntityEncoder[ExportSdtmDatasetVariablesRow] = jsonEncoderOf[ExportSdtmDatasetVariablesRow]
}
